File: 0226-InvertBinaryTree/Solution.java
import java.util.*;

public class Solution {
    
    public TreeNode invertTree(TreeNode root) {
        if (root == null) {
            return null;
        } else {
            TreeNode left = invertTree(root.right);
            TreeNode right = invertTree(root.left);
            return new TreeNode(root.val, left, right);
        }
    }

    public class TreeNode {
        int val;
        TreeNode left;
        TreeNode right;
        TreeNode() {}
        TreeNode(int val) { this.val = val; }
        TreeNode(int val, TreeNode left, TreeNode right) {
            this.val = val;
            this.left = left;
            this.right = right;
        }
    }

    public static void main(String[] args) {
        System.out.println("too cumbersome to test...");
    }

}

File: 0226-InvertBinaryTree/solution.py
from __future__ import annotations


class Solution:

    invert_tree(self, root: TreeNode) -> TreeNode:
        if not root:
            return None
        else:
            left = self.invert_tree(root.right)
            right = self.invert_tree(root.left)
            return TreeNode(root.val, left, right)


class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


if __name__ == "__main__":
    s = Solution()

